File: Source/Api/Organization/HierarchicalOrganization.cs
using System;
using System.Collections.Generic;
using ServiceRegister.Common;

namespace ServiceRegister.Api.Organization
{
    public class HierarchicalOrganization
    {
        public Guid Id { get; set; }
        public IEnumerable<LocalizedText> Names { get; set; }
        public IEnumerable<HierarchicalOrganization> SubOrganizations { get; set; }
    }
}

File: Source/AngularApplication/App/Models/HierarchicalOrganization.ts
"use strict";

module ServiceRegister
{
    export class HierarchicalOrganization extends OrganizationName implements Affecto.Base.IModel
    {
        public name: string;

        constructor(id: string, names: Array<LocalizedText>, public subOrganizations: Array<HierarchicalOrganization>)
        {
            super(id, names);
        }
    }
}  